### PR TITLE
Enforce more eslint rules

### PR DESCRIPTION
**Pull request by xPaw**
More consistent style, and avoid potential bugs.

### CODE REVIEW

**Comment by astorije** [comment]
Builds for this PR should pass after #684 is merged. I'll rebase this PR accordingly.

File: .eslintrc.yml [hunk 1]
```diff
@@ -11,27 +11,34 @@ parserOptions:
...
+  handle-callback-err: 2
   indent: [2, tab]
   key-spacing: [2, {beforeColon: false, afterColon: true}]
   keyword-spacing: [2, {before: true, after: true}]
   linebreak-style: [2, unix]
   no-console: 0
   no-control-regex: 0
-  no-inner-declarations: 2
-  no-invalid-regexp: 2
-  no-irregular-whitespace: 2
```
Comment (astorije) [inline .eslintrc.yml]: Why removing these 3? It looks to me like we do want to keep checking on them, no?

File: .eslintrc.yml [hunk 2]
```diff
@@ -11,27 +11,34 @@ parserOptions:
...
+  no-else-return: 2
+  no-implicit-globals: 2
+  no-multi-spaces: 2
+  no-shadow: 2
+  no-template-curly-in-string: 2
   no-trailing-spaces: 2
-  no-unexpected-multiline: 2
```
Comment (astorije) [inline .eslintrc.yml]: We might also want to keep this. From http://eslint.org/docs/rules/no-unexpected-multiline:

> Note that the patterns considered problems are not flagged by the semi rule.

File: .eslintrc.yml [hunk 3]
```diff
@@ -11,27 +11,34 @@ parserOptions:
...
-  no-unreachable: 2
```
Comment (astorije) [inline .eslintrc.yml]: Was that removed on purpose? Why?

File: client/js/lounge.js [hunk 4]
```diff
@@ -494,73 +495,75 @@ $(function() {
 		userStyles: userStyles.text(),
 	}, JSON.parse(window.localStorage.getItem("settings")));
 
-	for (var i in options) {
-		if (i === "userStyles") {
-			if (!/[\?&]nocss/.test(window.location.search)) {
-				$(document.head).find("#user-specified-css").html(options[i]);
+	(function SettingsScope() {
```
Comment (astorije) [inline client/js/lounge.js]: `Settings` with an upper case on purpose? Also, why wrapping these 70 lines?

File: scripts/build-fontawesome.js [hunk 5]
```diff
@@ -9,13 +9,13 @@ var fonts = [
 	"fontawesome-webfont.woff2"
 ];
 
-fs.ensureDir(destDir, function (err) {
-	if (err) {
-		console.error(err);
+fs.ensureDir(destDir, function(dirErr) {
+	if (dirErr) {
+		console.error(dirErr);
```
Comment (astorije) [inline scripts/build-fontawesome.js]: Why that change?

File: scripts/build-fontawesome.js [hunk 6]
```diff
@@ -9,13 +9,13 @@ var fonts = [
...
 	}
 
-	fonts.forEach(function (font) {
-		fs.copy(srcDir + font, destDir + font, function (err) {
+	fonts.forEach(function(font) {
+		fs.copy(srcDir + font, destDir + font, function(err) {
```
Comment (astorije) [inline scripts/build-fontawesome.js]: Urgh, I know this is to be consistent with the rest of our code, but this is painful to look at 😅. Sorry I missed this.
Also, I believe I should have used `=>` functions there instead...

File: src/client.js [hunk 7]
```diff
@@ -278,6 +278,10 @@ Client.prototype.updateToken = function(callback) {
 	var client = this;
 
 	crypto.randomBytes(48, function(err, buf) {
+		if (err) {
+			throw err;
+		}
```
Comment (astorije) [inline src/client.js]: Any unexpected consequences with this?

File: .eslintrc.yml (hunk 1 above)
Comment (xPaw) [inline .eslintrc.yml]: Theyre defaults in the recommended rules.

File: client/js/lounge.js (hunk 4 above)
Comment (xPaw) [inline client/js/lounge.js]: I wrapped them to reduce the scope (especially the I variable which was being shadowed in other functions)

File: src/client.js (hunk 7 above)
Comment (xPaw) [inline src/client.js]: It *should* crash.

File: scripts/build-fontawesome.js (hunk 5 above)
Comment (xPaw) [inline scripts/build-fontawesome.js]: Shadows same var name.

File: .eslintrc.yml (hunk 1 above)
Comment (astorije) [inline .eslintrc.yml]: Argh, I forgot to check that...

**Pull request merged**